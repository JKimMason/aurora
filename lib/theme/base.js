export default {
  colors: {
    // Main colors
    primary: "#3498db",
    success: "#2ecc71",
    danger: "#e74c3c",

    // Background
    background: "#F3F3F3",
    secondaryBackground: "white",

    // Lines and borders
    border: "#E5E5E5",

    // Text
    lightText: "#777777",
    text: "#282828"
  },

  font: "Helvetica",

  spacing: {
    padding: "1em"
  }
};
