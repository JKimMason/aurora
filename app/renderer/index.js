import React from "react";
import ReactDOM from "react-dom";
import Electron from "./components/Electron.js";
import Core from "../../lib/core";
import { Store } from "../../lib/mutate";

const render = Page => {
  ReactDOM.render(
    <Electron>
      <Page />
    </Electron>,
    document.getElementById("root")
  );
};

export default {
  index: () => {
    render(Core);
  },
  mutations: () => {
    render(Store);
  }
};
