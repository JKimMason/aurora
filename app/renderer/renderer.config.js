/**
 * Webpack config for compiling renderer
 */

const path = require("path");

module.exports = {
  context: path.resolve(__dirname),
  entry: ["babel-polyfill", "./index.js"],
  output: {
    path: path.resolve("app/.compiled"),
    filename: "renderer.bundle.js",
    library: "app",
    libraryTarget: "commonjs2"
  },
  target: "electron-renderer",
  module: {
    loaders: [
      {
        test: /\.js$/,
        loader: "babel-loader",
        exclude: /node_modules/,
        options: {
          presets: ["env", "react", "stage-2"]
        }
      },
      {
        test: /\.jsx$/,
        loader: "babel-loader",
        exclude: /node_modules/,
        options: {
          presets: ["env", "react", "stage-2"]
        }
      }
    ]
  }
};
