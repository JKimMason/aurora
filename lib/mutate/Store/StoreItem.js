import React from "react";
import { Card } from "../../ui";
import { PrimaryButton } from "../../ui/Buttons";
import styled from "styled-components";
import PropTypes from "prop-types";

const Title = styled.h2`
  margin-top: 0;
  font-weight: 300;
`;

const StoreItem = ({ title, description }) => (
  <Card>
    <Title>{title}</Title>
    <p>{description}</p>
    <PrimaryButton>Install</PrimaryButton>
  </Card>
);

StoreItem.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired
};

export default StoreItem;
