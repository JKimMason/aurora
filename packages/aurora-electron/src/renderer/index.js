import React from "react";
import ReactDOM from "react-dom";
import Core from "../../../aurora-core/index.js";

export default () => {
  ReactDOM.render(<Core />, document.getElementById("root"));
};
