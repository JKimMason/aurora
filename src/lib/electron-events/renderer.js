import names from "./names";
import { ipcRenderer } from "electron";

/**
 * Senders
 */
function sendInstallMutation(name) {
  ipcRenderer.send(names.INSTALL_MUTATION, name);
}

/**
 * Listeners
 */
function onChangeScreen(then) {
  return ipcRenderer.on(names.CHANGE_SCREEN, then);
}

function onMutationInstalled(then) {
  return ipcRenderer.on(names.INSTALL_MUTATION_REPLY, then);
}

export default {
  onChangeScreen,
  onMutationInstalled,
  sendInstallMutation
};
