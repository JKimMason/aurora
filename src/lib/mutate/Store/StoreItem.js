import React from "react";
import { Card } from "../../ui";
import { PrimaryButton, Button } from "../../ui/Buttons";
import { withShadow } from "../../ui/Modifiers";
import styled from "styled-components";
import PropTypes from "prop-types";
import { INSTALLING, INSTALLED, ERROR } from "./InstallStates";

const Title = styled.h2`
  margin-top: 0;
  font-weight: 300;
`;

const CardWithShadow = withShadow(Card);

const InstallButton = ({ installState, onClick }) => {
  switch (installState) {
    case INSTALLING:
      return (
        <PrimaryButton disabled onClick={onClick}>
          Install
        </PrimaryButton>
      );
    case INSTALLED:
      return <p> Installed! </p>;
    case ERROR:
      return <p> Error! </p>;
    default:
      return <PrimaryButton onClick={onClick}>Install</PrimaryButton>;
  }
};

const StoreItem = ({ title, description, onClick, installState }) => (
  <CardWithShadow>
    <Title>{title}</Title>
    {description && <p>{description}</p>}

    <InstallButton installState={installState} onClick={onClick} />
  </CardWithShadow>
);

StoreItem.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
  onClick: PropTypes.func.isRequired
};

export default StoreItem;
