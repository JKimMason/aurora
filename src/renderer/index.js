import React from "react";
import ReactDOM from "react-dom";
import ScreenManager from "./ScreenManager";
import { Provider } from "react-redux";
import reducers from "../redux/reducers";
import { setPreferences } from "../redux/actions";
import { createStore } from "redux";
import { loadPreferences } from "../lib/io";
import Draggable from "../lib/ui/Draggable";

const store = createStore(reducers);

class Aurora extends React.Component {
  constructor(props) {
    super(props);

    loadPreferences().then(prefs => {
      store.dispatch(setPreferences(prefs));
    });
  }

  render() {
    return (
      <Provider store={store}>
        <Draggable dragRegionSize="50px">
          <ScreenManager />
        </Draggable>
      </Provider>
    );
  }
}

// Add Electron to the page
document.body.innerHTML = `<div id="root"></div>`;
ReactDOM.render(<Aurora />, document.getElementById("root"));
