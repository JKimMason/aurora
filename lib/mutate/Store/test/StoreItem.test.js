import StoreItem from "../StoreItem";
import React from "react";
import { withTheme } from "../../../theme";
import renderer from "react-test-renderer";
import sinon from "sinon";
import { mount } from "enzyme";

const StoreItemInContext = withTheme(StoreItem);

test("StoreItem is defined", () => expect(StoreItemInContext).toBeDefined());

describe("StoreItem", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <StoreItemInContext
          title={"title"}
          description={"description"}
          onClick={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls onClick when the button is clicked", () => {
    const onClickSpy = sinon.spy();
    const wrapper = mount(
      <StoreItemInContext
        title={"title"}
        description={"description"}
        onClick={onClickSpy}
      />
    );

    wrapper.find("button").simulate("click");
    expect(onClickSpy.called).toBe(true);
  });
});
