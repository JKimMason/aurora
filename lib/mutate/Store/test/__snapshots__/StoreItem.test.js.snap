// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreItem renders correctly 1`] = `
<div
  className="sc-bdVaJa kDnqCF"
>
  <div
    className="sc-iwsKbI lhbiTu"
  >
    <h2
      className="sc-dnqmqq hrMvoW"
    >
      title
    </h2>
    <p>
      description
    </p>
    <button
      className="sc-bZQynM cEiCWc"
      onClick={[Function]}
    >
      Install
    </button>
  </div>
</div>
`;
